urlextract/urlextract_core.py

```diff
@@ -390,8 +395,18 @@ def _complete_url(
                 if start_pos <= 0:
                     left_ok = False
                 else:
-                    if text[start_pos - 1] not in self._stop_chars_left:
+                    left_char = text[start_pos - 1]
+                    if left_char not in self._stop_chars_left:
                         start_pos -= 1
+                        # extract the protocol if we've reached ":"
+                        if left_char == self._stop_protocol_char:
+                            for protocol in self._supported_protocols:
+                                protocol_start_pos = start_pos - len(protocol)
+                                if protocol_start_pos < 0:
+                                    continue
+                                if protocol == text[protocol_start_pos : start_pos]:
```

Comment:
I did not tried it but my guess is that this might not correctly extract URL which is valid and specified as this:
`svn+ssh://example.com`

Reply:
all the exiting unit tests passed, was this case in the unit tests?

Reply:
No, it is not there yet. This is just one that I can imagine that might have issues with this part of code. 

And as you can see below some test did not pass: https://github.com/lipoja/URLExtract/runs/5550973946?check_suite_focus=true - I did not check it deeply so I am not sure what caused it.